File: 0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {

        int result[]=new int[nums1.length];

        Map<Integer,Integer>map=new HashMap<>();
        Stack<Integer>stack=new Stack<>();


        for(int num:nums2)
        {
            while(!stack.isEmpty() && num>stack.peek())
            {
              map.put(stack.pop(),num);
            }
            stack.push(num);
        }

        while(!stack.isEmpty())
        {
            map.put(stack.pop(),-1);
        }

        for(int i=0;i<nums1.length;i++)
        {
            result[i]=map.get(nums1[i]);
        }

        return result;
        
    }
}

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        stack=[]
        mapp={}
        result=[]
        for i in range(len(nums2)):
            while stack and nums2[i]>stack[-1]:
                mapp[stack.pop()]=nums2[i]
            stack.append(nums2[i])
        
        while stack:
            mapp[stack.pop()]=-1
        
        for i in range(len(nums1)):
            result.append(mapp[nums1[i]])
            

        
        print(result)

        return result


        